torchvision/datasets/usps.py

```diff
@@ -57,8 +57,8 @@ def __init__(
         import bz2
         with bz2.open(full_path) as fp:
             raw_data = [line.decode().split() for line in fp.readlines()]
-            imgs = [[x.split(':')[-1] for x in data[1:]] for data in raw_data]
-            imgs = np.asarray(imgs, dtype=np.float32).reshape((-1, 16, 16))
+            tmp_list = [[x.split(':')[-1] for x in data[1:]] for data in raw_data]
```

Comment:
This is typically an instance of perfectly fine code that `mypy` flags as unsafe and it leads to obfuscated code.

Maybe we can just avoid the tmp variable and directly declare

```py
imgs = np.asarray(
	[...],
	dtype=...
)
```

Reply:
Mypy marks the code because `imgs` is originally defined as a list of lists and then it becomes a numpy. 

Not sure if inlining the expression increases readability or makes the code less obfuscated. Line 61 does already casting and reshaping, should we add splitting and iterating?